hpo_demo_setup.sh

```diff
@@ -0,0 +1,288 @@
+			HPO_DOCKER_IMAGE=${HPO_DOCKER_REPO}:${HPO_VERSION}
+		fi
+		if [[ ${CLUSTER_TYPE} == "native" ]]; then
+			echo
+			echo "Starting hpo with  ./deploy_hpo.sh -c ${CLUSTER_TYPE}"
+			echo
+			./deploy_hpo.sh -c ${CLUSTER_TYPE} >> ${LOGFILE}  &
+			#check_err "ERROR: HPO failed to start, exiting"
+		else
+			echo
+			echo "Starting hpo with  ./deploy_hpo.sh -c ${CLUSTER_TYPE} -h ${HPO_DOCKER_IMAGE}"
+			echo
+
+			./deploy_hpo.sh -c "${CLUSTER_TYPE}" -h "${HPO_DOCKER_IMAGE}"
+			#check_err "ERROR: HPO failed to start, exiting"
+		fi
+	popd >/dev/null
+	echo "#######################################"
+	echo
+}
+
+###########################################
+#   Start HPO Experiments
+###########################################
+function hpo_experiments() {
+
+	SEARCHSPACE_JSON="hpo_helpers/search_space.json"
+	URL="http://localhost:8085"
+	## TODO : Add check to eid
+	eid=$(${PY_CMD} -c "import hpo_helpers.utils; hpo_helpers.utils.getexperimentid(\"${SEARCHSPACE_JSON}\")")
+	exp_output="experiment-output.csv"
+
+	##TODO : Delete the old logs if any before starting the experiment
+
+	echo "#######################################"
+	echo "Start a new experiment with search space json"
+	## Step 1 : Start a new experiment with provided search space.
+	exp_json=$( cat ${SEARCHSPACE_JSON} )
+	if [[ ${exp_json} != "" ]]; then
+		curl -Ss -H 'Content-Type: application/json' ${URL}/experiment_trials -d '{ "operation": "EXP_TRIAL_GENERATE_NEW",  "search_space": '"${exp_json}"'}'
+		check_err "Error: Creating the new experiment failed."
+	else
+		check_err "Error: Searchspace is empty"
+	fi
+
+	## Looping through trials of an experiment
+	for (( i=0 ; i<${N_TRIALS} ; i++ ))
+	do
+		## Step 2: Get the HPO config from HPOaaS
+		echo "#######################################"
+        	echo "Generate the config for trial ${i}"
+		sleep 2
+		HPO_CONFIG=$(curl -Ss -H 'Accept: application/json' "${URL}"'/experiment_trials?experiment_id='"${eid}"'&trial_number='"${i}")
+		#echo ${HPO_CONFIG}
+		if [[ ${HPO_CONFIG} != -1 ]]; then
+			echo "${HPO_CONFIG}" > hpo_config.json
+		else
+			check_err "Error: Issue generating the configuration from HPO."
+		fi
+
+		## Step 3: Run the benchmark with HPO config.
+		echo "#######################################"
+	        echo "Run the benchmark for trial ${i}"
+		BENCHMARK_OUTPUT=$(./hpo_helpers/runbenchmark.sh ${SEARCHSPACE_JSON})
+		#BENCHMARK_OUTPUT="Objfunc_result=0.007914818407446147 Benchmark_status=prune"
+		echo ${BENCHMARK_OUTPUT}
+		obj_result=$(echo ${BENCHMARK_OUTPUT} | cut -d "=" -f2 | cut -d " " -f1)
+		trial_state=$(echo ${BENCHMARK_OUTPUT} | cut -d "=" -f3 | cut -d " " -f1)
+		if [[ ${obj_result} == "" || ${trial_state} == "" ]]; then
+			obj_result=0
+			trial_state="prune"
+		fi
+		### Add the HPO config and output data from benchmark of all trials into single csv
+		${PY_CMD} -c "import hpo_helpers.utils; hpo_helpers.utils.hpoconfig2csv(\"hpo_config.json\",\"output.csv\",\"${exp_output}\",\"${i}\")"
```

Comment:
This seems out of place here. Please create a function for this and move it there. This function can be called optionally to illustrate its usage.